client/layout/index.jsx

```diff
@@ -147,6 +147,23 @@ class Layout extends Component {
 	}
 
 	render() {
+		// This is temporary helper function until we have rolled out to 100% of customers.
+		const isNavUnificationEnabled = () => {
```

Comment:
Is this the right place for this logic? Several things hint that it's not, but let's talk it through and decide. I have some thoughts and would love to hear your thoughts about it.

1. It seems like `isAutomatticTeamMember` relies on the `QueryReaderTeams` triggering a fetch, then receiving that data. This means that if we open `wordpress.com` as a logged-in user, but with a clean Redux store (try it out yourself), we'll see nav unification always disabled. Is that the expected behavior? This also has several sub-questions:
* If we move the check earlier, won't we have the same problem because the teams' data will still not be fetched?
* If we move it later, won't it cause a weird experience when it loads as non-nav-unification and changes the nav to the nav unification one later?
* In any case, this behavior should be consistent and should yield the same results, regardless of whether we're using it with a clean Redux store or not.
2. This currently happens inside the `<Layout />` component, and I'm not sure this is the right place for it. I realize it controls visual changes, so having it in the layout component makes sense, but at the same time, this changes a `config` var. Anything that was called prior to the rendering of `Layout` will not use the proper value, and that might cause unexpected behavior.
3. There are multiple locations where we call just `isEnabled( 'nav-unification' )` and rely on it. Can we somehow guarantee all of those instances are called **after** this logic? Otherwise, we're poised to receiving inconsistent results.
4. Currently, logged out experience displays the old navigation. At the same time, after we login, we see the new one. Have we considered the implications of this difference in navigations after we've merged this for all users? Will all users have to switch from one nav to another? This feels like a regression to me.
5. Minor: this function is declared inside a method and will be redefined on every re-render. Do we want this, or should we declare it inside its own method in the component?

Reply:
> 1. [...] we'll see nav unification always disabled. Is that the expected behavior?

I think this it ok for now. We are not interested on making sure that all a12s get a consistent nav experience always, we are only interested on getting feedback about the new nav experience. Of course we're aiming for enabling the new nav experience every time that it's possible, but we can live with some edge cases in where the old nav experience will be incorrectly shown (as long as they are edge cases and very unlikely to happen).

> 2. having it in the layout component makes sense, but at the same time, this changes a `config` var
> 3. There are multiple locations where we call just `isEnabled( 'nav-unification' )` and rely on it

[I also have concerns about dynamically changing the `config` vars](https://github.com/Automattic/wp-calypso/pull/49136#discussion_r563780120). I would consider revisiting the approach and implement something that does not rely on `config` vars since that will simplify things a log. @cpapazoglou I'm happy to push an update looking into this direction if you don't mind.

> 4. Currently, logged out experience displays the old navigation

Probably we're missing a scenario here. How a logged out user can see any navigation at all? We thought that the navigation (either new or old) is only displayed to logged in users.

> 5. this function is declared inside a method and will be redefined on every re-render.

Good catch. We'll make sure it's declared outside.

Reply:
In that case, with this in mind and https://github.com/Automattic/wp-calypso/pull/49136#discussion_r564505110 I'm fine with this as a temporary solution to make testing for a12s easier, more reliable while still keeping it possible to opt-out. 👍

Reply:
> Minor: this function is declared inside a method and will be redefined on every re-render. Do we want this, or should we declare it inside its own method in the component?

I have moved the function outside of `render`.

> Currently, logged out experience displays the old navigation. At the same time, after we login, we see the new one. Have we considered the implications of this difference in navigations after we've merged this for all users? Will all users have to switch from one nav to another? This feels like a regression to me.

This is inevitable till we roll out to 100% and thus remove any conditionals. Currently we need a user to determine whether the feature is enabled or not.

> There are multiple locations where we call just isEnabled( 'nav-unification' ) and rely on it. Can we somehow guarantee all of those instances are called after this logic? Otherwise, we're poised to receiving inconsistent results.

All components using `isEnabled( 'nav-unification' )` are currently descendants of `layout`

> It seems like isAutomatticTeamMember relies on the QueryReaderTeams triggering a fetch, then receiving that data. This means that if we open wordpress.com as a logged-in user, but with a clean Redux store (try it out yourself), we'll see nav unification always disabled. Is that the expected behavior? 

This should be fixed now. I was disabling the feature even when `this.props.teams.length === 0` resulting in bailing out early. Make sure you stop your process and restart by `yarn build && yarn start` as it heavily depends on the `development.json` which now defaults to true.